[Protocol Bytes] - Support to client 12.85.11525

marcosvf132 opened the pull request:
# Protocol 12.85

Co-author: @Beats-dh and @dudantas

## Description

![image](https://user-images.githubusercontent.com/61994374/148670538-40d16325-211b-402e-8a8a-e0702069950d.png)

- Update to protocol 12.85

- Credits for @hyresu and @Zbizu for helping on market bytes 12.81!

marcosvf132 committed:
Update item_parse.cpp

- Adding new items.xml tag on item.

marcosvf132 committed:
Update items_definitions.hpp

- Registering 'upgrade classification' parse label.

marcosvf132 committed:
Update items.h

- Register attribute on ItemType.

marcosvf132 committed:
Update protocolgame.cpp

- Adding new potocols bytes related to:

> Player skills.

> Item sender.

marcosvf132 committed:
Update item_parse.hpp

- New item attribute definition on parse.

marcosvf132 committed:
Update items.xml

- Insert 'upgradeclassification' atttribute on specific items.
- Update items values following client 12.81 assets and Tibia WIKI.
- Small typo fixes.

marcosvf132 committed:
Update protocolgame.cpp

- **Cyclopedia character**

- Insert provisory 12.72 missing bytes.

- Insert 12.81 missing bytes.

Zbizu commented:
you need a new byte in set outfit window (probably at the end of packet): randomize mount - checkbox (probably boolean)

I don't know if other outfit window modes also require that.

marcosvf132 committed:
Missing protocols and config version.

- Config.lua protocol version.

- Market missing bytes.

- Outfit window missing bytes.

marcosvf132 committed:
SonarCloud && Outfit window

- Change outfit window missing byte from uint16_t to uint8_t.

- SonarCloud hints.

marcosvf132 committed:
Update protocolgame.cpp

- Market enter missing byte.

marcosvf132 committed:
Pointer fix

Ops

marcosvf132 committed:
[Item classification] New class and protocol data

- Enable item classifications and tiers on Market and Cyclopedia.

marcosvf132 committed:
Cleaning and build fix

- Cleaning lua empty lines.
- Rename cpp include name.

marcosvf132 committed:
[Imbue] - Missing byte

marcosvf132 committed:
Github actions hints.

marcosvf132 committed:
- Market prices declaration

- On 12.81 protocol, market prices changed from u32 to u64.

marcosvf132 committed:
Update protocolgame.cpp

- Fix hint by Hyresu and Zbizu.

Zbizu commented on src/server/network/protocol/protocolgame.cpp:
```diff
@@ -4086,15 +4129,20 @@ void ProtocolGame::sendMarketBrowseItem(uint16_t itemId, const MarketOfferList &
 	NetworkMessage msg;
 
 	msg.addByte(0xF9);
+	msg.addByte(0);
```
use value 3 here

market got new enums and uses byte for request/response now instead of u16:
```
MARKETREQUEST_OWN_HISTORY = 1,
MARKETREQUEST_OWN_OFFERS = 2,
MARKETREQUEST_ITEM = 3,
```

marcosvf132 committed:
[Fix] - Server internal check was blocking movement event on items

marcosvf132 committed:
Merge branch 'master' into protocol-12.81

marcosvf132 committed:
Items.xml updated and few changes on protocolgame

marcosvf132 committed:
[Build] - Small fix

marcosvf132 committed:
Config and protocol fixes

marcosvf132 committed:
- Character info (cyclopedia)

marcosvf132 committed:
Items XML and OTB file

beats-dh committed:
12.85

beats-dh committed:
Merge branch 'master' into pr/167

beats-dh committed:
Update items_definitions.hpp

beats-dh committed:
fix build

beats-dh committed:
update

beats-dh committed:
Update outfits.xml

dudantas committed:
Enhancement of item tier classifications

dudantas committed:
Fix typo

dudantas committed:
Adjust

dudantas committed:
Enhancement of item tier classifications script

dudantas committed:
Merge branch 'protocol-12.81' of https://github.com/marcosvf132/canary into pr/167

dudantas committed:
Merge branch 'master' into pr/167
